src/main/java/io/vertx/mqtt/impl/MqttServerImpl.java

```diff
@@ -156,5 +191,17 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
         }
       }
     });
+
+    if(options.isUseWebsocket()) {
+
+      pipeline.addBefore("mqttEncoder", "httpServerCodec", new HttpServerCodec());
+      pipeline.addAfter("httpServerCodec", "aggregator", new HttpObjectAggregator(65536));
```

Comment:
Any specific reason you need this bigger 64k buffer?

Reply:
Actually 65536 is a max content length.

The aggregator will accept HTTP message until the request is ended and produce a `FullHttpRequest` object with a body. When the aggregation grows above the max content length then this is considered as an error, the client will receive an error response and connection is closed.

So this is a setting to prevent attacks of client that would be aware of this mechanism and try to OOM the JVM using it. It does not the allocated a buffer.